R/utils-json.R

```diff
@@ -278,10 +281,51 @@ subsetByMonth <- function(pts, onlyMonth) {
   }
   return(pts)
 }
-getYvals_approvals <- function(object, num_vals){
-  ylim <- ylim(object)$side.2[1]
-  yvals <- rep(ylim, num_vals)
-  return(yvals)
+
+ApprovalInfoY <- function(object, times, c) {
+  # Compute top and bottom vertical position of approval bars.
+  # 
+  # Args:
+  #   object: A gsplot object.
+  #   times: "times" argument of R rep() function.
+  #   c: A scaling ratio to adjust top or bottom of approval bar rectangle.
+  #
+  # Returns:
+  #   Approval bar, top or bottom y-axis point, in world coordinates.
+  reverse <- object$side.2$reverse
+  
+  # no idea whether semantics of object$global$par$ylog are correct in this
+  # context or not; it just looked promising in the debugger
+  if (is.null(object$global$par$ylog)) {
```

Comment:
yes, `object$global$par$ylog` is good - but I don't think it will ever be NULL based on what the `isLogged` function returns. I wonder if we should not have logic correcting for when this is NULL so that we can catch that scenario as a bug if it ever happens